lib/promoted/ruby/client/request_builder.rb

```diff
@@ -99,10 +103,11 @@ def log_request_params(include_insertions: true, include_request: true)
             client_info: @client_info
           }
 
-          request[:request_id] = @id_generator.newID if not request[:request_id]
-
           # Log request allows for multiple requests but here we only send one.
-          params[:request] = [request] if include_request
+          if include_request
+            request[:request_id] = request[:request_id] || @id_generator.newID
```

Comment:
It's hard for me to tell with ruby client if this covers all of the cases that I'd expect

Reply:
ftr in practice, this is the "SDK-did-delivery" case.